e2e-tests-rust/tests/lib.rs

```diff
@@ -95,3 +102,19 @@ async fn interval_sealing_multiple_txs() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn no_sealing_timeout() -> anyhow::Result<()> {
+    // Test that we can submit a transaction and timeout while waiting for it to finalize when
+    // era-test-node is operating in no sealing mode.
+    let provider = init(|node| node.no_mine()).await?;
+
+    let tx = TransactionRequest::default()
+        .with_to(address!("d8dA6BF26964aF9D7eEd9e03E53415D37aA96045"))
+        .with_value(U256::from(100));
+    let pending_tx = provider.send_transaction(tx).await?.register().await?;
+    let finalization_result = tokio::time::timeout(Duration::from_secs(3), pending_tx).await;
+    assert!(finalization_result.is_err());
```

Comment:
Can we force block to seal? If so, it would be nice to add it to this test too to make sure that the node doesn't just hang

Reply:
We can seal a block manually but it won't have the tx in it (see https://github.com/matter-labs/era-test-node/issues/404). I am hoping to address this shortly and will update this test then